Fixed Sidetone Gain not immediately applied after change / Canary Move / warn refactor

db4ple opened the pull request:
- Now in all modes which have side tone, the change of side tone volume
  is applied immediately (before only in Tune).
- The extracted canary methods are now used.
- added a variable init to keep the compiler silent in freedv_api

m-chichikalov commented on mchf-eclipse/src/uhsdr_main.c:
```diff
@@ -46,6 +46,7 @@
 #include "drivers/ui/oscillator/osc_si570.h"
 #include "drivers/audio/codec/codec.h"
 #include "misc/profiling.h"
+#include "uhsdr_canary.h"
```
You could remove including malloc.h up there also, I think...

db4ple replied:
Done.

db4ple committed:
Fixed Sidetone Gain not immediately applied after change / Canary Move / warn refactor
- Now in all modes which have side tone, the change of side tone volume
  is applied immediately (before only in Tune).
- The extracted canary methods are now used.
- added a variable init to keep the compiler silent in freedv_api
